### ISSUE TITLE
[Question] Recyclerview of Conductor Controllers

### ISSUE THREAD

**Issue by kikin81** [bluelinelabs/Conductor#368]
I wanted to know if anyone had attempted to do a recycler view in which each item was a conductor controller and the lifecycle repercussions which might arise.
I have a pretty complicated UI that displays multiple conductor controllers. Originally it was written using a ScrollView with multiple FrameLayouts for the controller containers.
The scrollview performance was not that great and animating inserting/removing items was a pain.
I think of each conductor controller in this recyclerview as a "mvp widget" which is self contained. Some controllers will be similar (same view type) and others are static (header).
I have a prototype that is somewhat working. The individual controllers are getting instantiated correctly and running their logic. When a view is scrolled off screen the controller's `onDestroyView` is called. I am working on a way to restore the state now.

**Comment by yesitskev** [comment]
I was looking into something this in the past but haven't gotten around to implementation. If this ticket is still open when the time comes i'll dump my findings.

**Comment by sevar83** [comment]
Hi, @kikin81, @kevelbreh. I'm quite interested in your findings with controllers in RV. Did you succeed to recycle their views? What's your progress?

**Comment by sockeqwe** [comment]
I'm sorry to say that, but this approach sounds like a bad idea to me.

I have done something similar years ago (not conductor) and it worked out pretty bad.
The problem is that  the adapter data set is controlled by someone else then each single ViewHolder Controller. Then add orientation changes plus backstack navigation plus recycling and reusing of Views (and then ViewControllers somehow too?) to that equation and you get a ... mess.

It's much easier and simpler to just look at RecyclerView - ViewHolders as View layer and "controll" update it through the "parent" ViewController that holds the Adapter and RecyclerView. Use DiffUtils.

**Comment by sevar83** [comment]
Hi Hannes @sockeqwe , we spoke about this few months ago. I did it as you suggested. But my main problem now is with the RV. Probably I've overused/abused it. It's not designed to have very large item layouts. And my items became huge. Actually each item is a detail screen. My design idea was to have a scrollable feed like the facebook feed. When i started I didn't realize how difficult this will be. Later on, they introduced Litho to tackle with these problems. So, at first render the RV attempts to bind a lot of items (around 5), although only one is visible at once on the screen (like a ViewPager). It then unbinds 2 of them right away. Maybe because it decides it doesn't need too much of them, not sure. Waste of layout time! This internal logic is fairly complicated and I couldn't find a way to control how many holders to bind at first initialization. In this aspect `ViewPager` is much simpler. That's why I'm researching possibilities to migrate to ViewPager and MAYBE use controllers as items. Otherwise all changes to the items must flow as DiffUtil partial RV viewholder bindings (you know, with payloads). This mechanism is not very scaleable. In fact almost all my view logic is in the viewholders - not even in the controller holding the RV.
TLDR: FML :D
